PublishReadyToRun fails on Linux with exit code 139 when Microsoft.CodeAnalysis package referenced

**Steps to reproduce:**
1. git clone [https://github.com/plwm/eShopOnWeb-dotnetcore3.0](https://github.com/plwm/eShopOnWeb-dotnetcore3.0)
2. cd src/Web
2. Modify Web.csproj
-> uncomment `<PackageReference Include="Microsoft.CodeAnalysis" Version="3.2.1" />`
3.  dotnet publish -c Release -r linux-x64 -o out/linux-x64-3.0/

Build fails when publish on Linux with runtime linux-x64 or linux-arm
Build succeeds when publish on Windows with runtime win-x64

**Note:**
Build succeeds on Linux unless you uncomment: `<PackageReference Include="Microsoft.CodeAnalysis" Version="3.2.1" />`
Same behaviour with latest version 3.3.1-beta3-final of the library.

**Platform:**
```
$ uname -a
Linux AI509930 4.4.0-17134-Microsoft #706-Microsoft Mon Apr 01 18:13:00 PST 2019 x86_64 x86_64 x86_64 GNU/Linux

$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-rc1-014190
 Commit:    c4d43f672d

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  19.04
 OS Platform: Linux
 RID:         ubuntu.19.04-x64
 Base Path:   /home/marcin/dotnet/sdk/3.0.100-rc1-014190/

Host (useful for support):
  Version: 3.0.0-rc1-19456-20
  Commit:  8f5d7b1ba4

.NET Core SDKs installed:
  3.0.100-rc1-014190 [/home/marcin/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0-rc1.19457.4 [/home/marcin/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0-rc1-19456-20 [/home/marcin/dotnet/shared/Microsoft.NETCore.App]
```

**Log:**
```
$ dotnet publish -c Release -r linux-x64 -o out/linux-x64-3.0/
Microsoft (R) Build Engine version 16.3.0-preview-19455-02+4a2d77107 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 373.34 ms for /home/marcin/eShopOnWeb/src/ApplicationCore/ApplicationCore.csproj.
  Restore completed in 413.45 ms for /home/marcin/eShopOnWeb/src/Infrastructure/Infrastructure.csproj.
  Restore completed in 507.59 ms for /home/marcin/eShopOnWeb/src/Web/Web.csproj.
  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview

  Restore operation started...
  Restoring library jquery@3.3.1...
  Restoring library bootstrap@3.3.7...
  Restoring library jquery-validation-unobtrusive@3.2.10...
  Restoring library jquery-validate@1.17.0...
  Restoring library toastr.js@2.1.4...
  Restoring library @aspnet/signalr@1.0.3...

  6 libraries restored in 7.17 seconds

  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
  ApplicationCore -> /home/marcin/eShopOnWeb/src/ApplicationCore/bin/Release/netstandard2.1/ApplicationCore.dll
  Infrastructure -> /home/marcin/eShopOnWeb/src/Infrastructure/bin/Release/netstandard2.1/Infrastructure.dll

  Bundler: Begin processing bundleconfig.json
  Bundler: Done processing bundleconfig.json
  Web -> /home/marcin/eShopOnWeb/src/Web/bin/Release/netcoreapp3.0/linux-x64/Web.dll
  Web -> /home/marcin/eShopOnWeb/src/Web/bin/Release/netcoreapp3.0/linux-x64/Web.Views.dll
  Optimizing assemblies for size, which may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
/home/marcin/dotnet/sdk/3.0.100-rc1-014190/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(307,5): error MSB6006: "/home/marcin/.nuget/packages/microsoft.netcore.app.runtime.linux-x64/3.0.0-rc1-19456-20/tools/crossgen" exited with code 139. [/home/marcin/eShopOnWeb/src/Web/Web.csproj]
/home/marcin/dotnet/sdk/3.0.100-rc1-014190/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(307,5): error MSB6006: "/home/marcin/.nuget/packages/microsoft.netcore.app.runtime.linux-x64/3.0.0-rc1-19456-20/tools/crossgen" exited with code 139. [/home/marcin/eShopOnWeb/src/Web/Web.csproj]
/home/marcin/dotnet/sdk/3.0.100-rc1-014190/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(307,5): error MSB6006: "/home/marcin/.nuget/packages/microsoft.netcore.app.runtime.linux-x64/3.0.0-rc1-19456-20/tools/crossgen" exited with code 139. [/home/marcin/eShopOnWeb/src/Web/Web.csproj]
/home/marcin/dotnet/sdk/3.0.100-rc1-014190/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(307,5): error MSB6006: "/home/marcin/.nuget/packages/microsoft.netcore.app.runtime.linux-x64/3.0.0-rc1-19456-20/tools/crossgen" exited with code 139. [/home/marcin/eShopOnWeb/src/Web/Web.csproj]
/home/marcin/dotnet/sdk/3.0.100-rc1-014190/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(307,5): error MSB6006: "/home/marcin/.nuget/packages/microsoft.netcore.app.runtime.linux-x64/3.0.0-rc1-19456-20/tools/crossgen" exited with code 139. [/home/marcin/eShopOnWeb/src/Web/Web.csproj]
/home/marcin/dotnet/sdk/3.0.100-rc1-014190/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(219,5): error NETSDK1096: Optimizing assemblies for performance failed. You can either exclude the failing assemblies from being optimized, or set the PublishReadyToRun property to false. [/home/marcin/eShopOnWeb/src/Web/Web.csproj]
```

--- a/src/vm/genmeth.cpp
+++ b/src/vm/genmeth.cpp
@@ -802,7 +802,11 @@ MethodDesc::FindOrCreateAssociatedMethodDesc(MethodDesc* pDefMD,
     // Some callers pass a pExactMT that is a subtype of a parent type of pDefMD.
     // Find the actual exact parent of pDefMD.
     pExactMT = pDefMD->GetExactDeclaringType(pExactMT);
-    _ASSERTE(pExactMT != NULL);
+    if (pExactMT == NULL)
+    {
+        _ASSERTE(false);
+        COMPlusThrowHR(COR_E_TYPELOAD);
+    }
 
     if (pDefMD->HasClassOrMethodInstantiation() || !methodInst.IsEmpty())
     {

